### DISCUSSION TITLE
Why not just use JQ for JSON processing language?

### DISCUSSION THREAD

**Discussion by isaacsanders** [grafana/grafana-infinity-datasource#921] [Ideas]
I feel like there are 3 poorly supported options for languages to use for processing JSON data. Why not just use a more mature option like JQ?

**Comment by cboggs** [comment]
That, or even something like `go-jsonnet` (https://github.com/google/go-jsonnet). Very stable and exceptionally well-documented by comparison to jsonata and UQL. :-)

My main struggle with using jsonata is that it is not actually a released version of jsonata that is supported, but rather an undocumented fork of a [Go](https://github.com/blues/jsonata-go) binding of an undocumented and unsupported version of Jsonata (`1.5.4`) that implements some features from `>= 1.7.0`  (like `distinct`, which does _not_ exist in 1.5.4), but also doesn't support either [positional](https://docs.jsonata.org/path-operators#-positional-variable-binding) nor [context](https://docs.jsonata.org/path-operators#-context-variable-binding) variable binding, which is supported in `1.7.0` but causes panics in `blues/jsonata-go`.

It's a very unusable setup unfortunately, as it is today. :-(

**Reply by yosiasz** [reply]
`JQ` looks beautiful! Why not fork infinity and implement and propose that as an addition?

**Comment by yesoreyeram** [comment]
`jq` support is coming in 3.0.0 via https://github.com/grafana/grafana-infinity-datasource/pull/1143

**Reply by isaacsanders** [reply]
This is so exciting

**Reply by yesoreyeram** [reply]
@isaacsanders - Are you fancy about testing? I will update the issue with steps to test. But here are the quick steps how to setup /test. Once grafana running, make sure to select `jq backend` as the parser.

* `git clone github.com/grafana/grafana-infinity-datasource && cd grafana-infinity-datasource`
* `git checkout jq-parser && yarn && yarn build && mage -v && docker compose up`

**Reply by yesoreyeram** [reply]
@yosiasz ?
